app/components/utils/mandatee-selector.js

```diff
@@ -32,27 +41,23 @@ export default class MandateeSelector extends Component {
     if (this.initialLoad.isRunning) {
       yield this.initialLoad;
     }
-    const queryOptions = {
-      ...this.defaultQueryOptions,  // clone
-    };
-    if (searchTerm) {
-      queryOptions['filter[person][last-name]'] = searchTerm;
-    }
-    queryOptions['filter[government-body][:uri:]'] = CURRENT_GOVERNMENT_BODY;
+    let results = yield this.loadCurrentBodyMandatees(searchTerm);
 
-    const results = yield this.store.query('mandatee', queryOptions);
-    // Many versions of a mandatee exist within a government-body.
-    // We only want the mandatees with no end-date or an end-date in the future.
-    // mu-cl-resources doesn't have :has-no:-capability for properties.
-    return results.filter((mandatee) => {
-      return !mandatee.end || (mandatee.end > new Date());
-    });
+    const currentBodyStartDate = results.firstObject.start;
+    if (currentBodyStartDate > this.referenceDate) {
+      results = yield this.loadPastBodyMandatees(searchTerm);
+    }
+    return results;
```

Comment:
@erikap are you okay with the current order of calls here, given that a test case will be added for `loadCurrentBodyMandatees` and a warning is added? 

This fetches the current government composition first in all cases, since we don't know the start date for it beforehand. If the `referenceDate` is after this start date, we're done (will be the case most of the time).
If the `referenceDate` is before this start date, a new request is made to get the correct past government composition, of which we're sure it'll have an end date.

Advantages I see for this approach vs. [your previous suggestion](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1323/files/ad205a0838ef4a8321c4239101ac7b514760e3ec#r866659408):
- no need for a fallback if getting the government body based on the reference date fails due to the mandatees having no end property (would happen *a lot*, basically for every new agenda item).
- all filtering happens in the backend, which I assume is more efficient than in the browser.

Unless I'm missing something?

Reply:
I'm having difficulties with the mixture of current government composition and mandatee start/end date. We talk about government bodies in methods/variable names, but in the queries we only filter on mandatee dates. Mandatee and government composition (< government body) are separate resources which both have a start/end date. 

In the current implementation we fetch mandatees without an end-date first and assign the start date of the first result entry (which is a mandatee and not a government body) to `currentBodyStartDate`, but these dates are not necessarily the same. E.g. the current active ministers have a start date of 10/5/2021 while the start date of the current government composition is 02/10/2019.

Just renaming `loadCurrentBodyMandatees` to `loadCurrentActiveMandatees` and `loadPastBodyMandatees` to `loadPastMandatees` will resolve the confusion.

Thinking further the current implementation also assumes the start date is the same for all current active mandatees (the ones without end date), which is the case in the current data, but might not hold in the future. Maybe even the very near future with the resignation of minister Beke. We will land in a case where based on the reference date we need to offer an options list with a mixture of mandatees with **and** without end date. That will not work in the current implementation.

E.g. the mandatee of minister Beke gets an end date of 16/5/2022. If they modify the mandatees on an agendaitem of 29/4/2022 they will want to get a list of the current active mandatees, but also including the mandatee of minister Beke which was active at that moment.